functions/autopilotRedirect.protected.ts

```diff
@@ -1,40 +1,98 @@
-import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
-import { JsonObject } from 'swagger-ui-express';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
 
 export interface Event {
+  Channel: string;
+  CurrentTask: string;
   Memory: string;
+  UserIdentifier: string;
 }
 
-export const handler = (context: Context, event: Event, callback: ServerlessCallback) => {
+type EnvVars = {
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+const handleChatChannel = async (context: Context<EnvVars>, event: Event) => {
+  const memory = JSON.parse(event.Memory);
+  const { ServiceSid, ChannelSid } = memory.twilio.chat;
+
+  const channel = await context
+    .getTwilioClient()
+    .chat.services(ServiceSid)
+    .channels(ChannelSid)
+    .fetch();
+
+  const attributes = JSON.parse(channel.attributes);
+
+  // if channel is webchat, disable the input
+  if (attributes.channel_type === 'web') {
+    const user = await context
+      .getTwilioClient()
+      .chat.services(ServiceSid)
+      .users(event.UserIdentifier)
+      .fetch();
+
+    const userAttr = JSON.parse(user.attributes);
+    const updatedAttr = { ...userAttr, lockInput: true };
+
+    await user.update({ attributes: JSON.stringify(updatedAttr) });
+  }
+};
+
+const buildActionsArray = (context: Context<EnvVars>, event: Event) => {
   const memory = JSON.parse(event.Memory);
-  const returnObj: JsonObject = { actions: [] };
 
   switch (memory.at) {
-    case 'survey':
-    case 'gender_why':
-      // eslint-disable-next-line no-case-declarations
+    case 'survey': {
+      if (event.Channel === 'voice') {
+        const say = { say: 'Hold on, we are connecting you with an agent' };
+        const handoff = {
+          handoff: {
+            channel: 'voice',
+            uri: `taskrouter://${context.TWILIO_CHAT_TRANSFER_WORKFLOW_SID}`,
+          },
+        };
+        return [say, handoff];
+      }
+
```

Comment:
Since we are not doing voice for now, I think we should remove this.  But that is a suggestion only -- I'm open to keeping it.